protocol/synthetix/contracts/modules/core/VaultModule.sol

```diff
@@ -34,49 +36,65 @@ contract VaultModule is IVaultModule {
 
     bytes32 private constant _DELEGATE_FEATURE_FLAG = "delegateCollateral";
 
     /**
      * @inheritdoc IVaultModule
      */
-    function delegateCollateral(
+    function declareIntentToDelegateCollateral(
         uint128 accountId,
         uint128 poolId,
         address collateralType,
-        uint256 newCollateralAmountD18,
+        int256 deltaCollateralAmountD18,
         uint256 leverage
-    ) external override {
+    ) external override returns (uint256 intentId) {
+        // Ensure the caller is authorized to represent the account.
         FeatureFlag.ensureAccessToFeature(_DELEGATE_FEATURE_FLAG);
         Account.loadAccountAndValidatePermission(accountId, AccountRBAC._DELEGATE_PERMISSION);
 
-        // Each collateral type may specify a minimum collateral amount that can be delegated.
-        // See CollateralConfiguration.minDelegationD18.
-        if (newCollateralAmountD18 > 0) {
-            CollateralConfiguration.requireSufficientDelegation(
-                collateralType,
-                newCollateralAmountD18
-            );
-        }
-
+        // Input checks
         // System only supports leverage of 1.0 for now.
         if (leverage != DecimalMath.UNIT) revert InvalidLeverage(leverage);
+        // Ensure current collateral amount differs from the new collateral amount.
+        if (deltaCollateralAmountD18 == 0) revert InvalidCollateralAmount();
+
+        // Verify the account holds enough collateral to execute the intent.
+        // Get previous intents cache
+        AccountDelegationIntents.Data storage accountIntents = AccountDelegationIntents.getValid(
+            accountId
+        );
 
         // Identify the vault that corresponds to this collateral type and pool id.
         Vault.Data storage vault = Pool.loadExisting(poolId).vaults[collateralType];
 
-        // Use account interaction to update its rewards.
-        uint256 totalSharesD18 = vault.currentEpoch().accountsDebtDistribution.totalSharesD18;
-        uint256 actorSharesD18 = vault.currentEpoch().accountsDebtDistribution.getActorShares(
-            accountId.toBytes32()
-        );
-
         uint256 currentCollateralAmount = vault.currentAccountCollateral(accountId);
+        int256 accumulatedDelta = deltaCollateralAmountD18 +
+            accountIntents.netDelegatedAmountPerCollateral[collateralType];
+        if (accumulatedDelta < 0 && currentCollateralAmount < (-1 * accumulatedDelta).toUint()) {
+            revert ExceedingUndelegateAmount(
+                deltaCollateralAmountD18,
+                accountIntents.netDelegatedAmountPerCollateral[collateralType],
+                accumulatedDelta,
+                currentCollateralAmount
+            );
+        }
 
-        // Conditions for collateral amount
+        uint256 newCollateralAmountD18 = accumulatedDelta > 0
+            ? currentCollateralAmount + (accumulatedDelta).toUint()
+            : currentCollateralAmount - (-1 * accumulatedDelta).toUint();
```

Comment:
It's needed because currentCollateralAmount is an `uint` and accumulatedDelta is `int`. And at that point is negative, so we need to get the modulus and substract from the uint

Reply:
maybe a really dumb suggestion 😅  but isn't it possible to just perform the following: 
```
        uint256 newCollateralAmountD18 =  currentCollateralAmount - (-1 * accumulatedDelta).toUint();
```
Since either way the minus is cancelled?

Reply:
We need to use the two branches here since accumulatedDelta can be positive or negative and toUint() will revert if the number is negative. An alternative would be to do:
`uint256 newCollateralAmountD18 =  `(currentCollateralAmount.toInt() + accumulatedDelta).toUint();`